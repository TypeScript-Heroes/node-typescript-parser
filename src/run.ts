import { TypescriptParser } from './';

const parser = new TypescriptParser();

const code = `
class Foo {
    private _calc: string;

    public get calc(): string {
        return this._calc;
    }

    public set calc(value: string) {
        this._calc = value;
    }

    public foo(): void {

    }
}
`;

async function parse(): Promise<void> {
    // const parsed = await parser.parseSource(`import Foo, { Foobar } from 'myLib';`);
    // const parsed = await parser.parseSource(`import Foo from 'myLib';`);
    // const parsed = await parser.parseSource(`import { Foo } from 'myLib';`);
    // const parsed = await parser.parseSource(`import { Foo, default as Bar } from 'myLib';`);
    const parsed = await parser.parseSource(code);
    console.log(parsed);
}

parse().then(() => process.exit(0)).catch(() => process.exit(1));
