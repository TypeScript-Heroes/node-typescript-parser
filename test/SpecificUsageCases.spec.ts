import { TypescriptParser } from '../src';
import { getWorkspaceFile, rootPath } from './testUtilities';

describe('Specific usage cases', () => {
    const parser = new TypescriptParser();

    describe('i18next with and without destructure', () => {

        it('should contain i18next reference in not destructured way', async () => {
            const file = getWorkspaceFile('specific-usage-cases/i18next-destructure/import-my-component.tsx');
            const parsed = await parser.parseFile(file, rootPath);
            expect(parsed.usages).toMatchSnapshot();
        });

        it('should contain t reference in destructured way', async () => {
            const file = getWorkspaceFile('specific-usage-cases/i18next-destructure/destructure-my-component.tsx');
            const parsed = await parser.parseFile(file, rootPath);
            expect(parsed.usages).toMatchSnapshot();
        });

    });

});
