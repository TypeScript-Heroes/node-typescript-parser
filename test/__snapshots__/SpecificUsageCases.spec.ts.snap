// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Specific usage cases i18next with and without destructure should contain i18next reference in not destructured way 1`] = `
Array [
  "MyComponent",
  "p",
  "i18next",
]
`;

exports[`Specific usage cases i18next with and without destructure should contain t reference in destructured way 1`] = `
Array [
  "MyComponent",
  "p",
  "t",
]
`;
